msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -90,8 +101,50 @@ public Integer getNumberOfDataPoints() {
 
 	@Override
 	public double[] getMzValues() {
-		// TODO Auto-generated method stub
-		return null;
+		double[] result = null;
+		byte[] bytesIn = new byte[getMzBinaryDataInfo().getEncodedLength()];
+		int precision = 0;
+		EnumSet<MzMLBinaryDataInfo.MzMLCompressionType> compressions = EnumSet
+				.noneOf(MzMLBinaryDataInfo.MzMLCompressionType.class);
+		try {
+			mappedByteBufferInputStream.setPosition(getMzBinaryDataInfo().getPosition());
+			XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+			XMLEventReader xmlEventReader = xmlInputFactory.createXMLEventReader(mappedByteBufferInputStream);
+
+			if (getMzBinaryDataInfo().getBitLength().toString().startsWith("THIRTY_TWO")) {
```

Comment:
It returns the ENUM Constant, and we have to know whether it is a constant referring to a 32 bit or 64 bit using its declared name (Hence "toString()")